dashboard/client/src/pages/node/NodeRow.tsx

```diff
@@ -0,0 +1,231 @@
+   */
+  onExpandButtonClick: () => void;
+};
+
+/**
+ * A single row that represents the node information only.
+ * Does not show any data about the node's workers.
+ */
+const NodeRow = ({
+  node,
+  rowIndex,
+  expanded,
+  onExpandButtonClick,
+}: NodeRowProps) => {
+  const {
+    hostname = "",
+    ip = "",
+    cpu = 0,
+    mem,
+    disk,
+    networkSpeed = [0, 0],
+    raylet,
+    logUrl,
+  } = node;
+
+  const classes = useNodeRowStyles();
+
+  return (
+    <TableRow key={hostname + rowIndex}>
+      <TableCell>
+        <IconButton size="small" onClick={onExpandButtonClick}>
+          {!expanded ? (
+            <AddIcon className={classes.expandCollapseIcon} />
+          ) : (
+            <RemoveIcon className={classes.expandCollapseIcon} />
+          )}
+        </IconButton>
+      </TableCell>
+      <TableCell>
+        <StatusChip type="node" status={raylet.state} />
+      </TableCell>
+      <TableCell align="center">
+        <Tooltip title={raylet.nodeId} arrow interactive>
+          <Link to={`/node/${raylet.nodeId}`} className={classes.idCol}>
+            {raylet.nodeId}
+          </Link>
+        </Tooltip>
+      </TableCell>
+      <TableCell align="center">{hostname}</TableCell>
+      <TableCell align="center">{ip}</TableCell>
+      <TableCell>
+        <PercentageBar num={Number(cpu)} total={100}>
+          {cpu}%
+        </PercentageBar>
+      </TableCell>
+      <TableCell>
+        <PercentageBar num={Number(mem[0] - mem[1])} total={mem[0]}>
+          {memoryConverter(mem[0] - mem[1])}/{memoryConverter(mem[0])}(
+          {mem[2].toFixed(1)}
+          %)
+        </PercentageBar>
+      </TableCell>
+      <TableCell>
+        {raylet && (
+          <PercentageBar
+            num={raylet.objectStoreUsedMemory}
+            total={raylet.objectStoreAvailableMemory}
+          >
+            {memoryConverter(raylet.objectStoreUsedMemory)}/
+            {memoryConverter(raylet.objectStoreAvailableMemory)}
+          </PercentageBar>
+        )}
+      </TableCell>
+      <TableCell>
+        {disk && disk["/"] && (
+          <PercentageBar num={Number(disk["/"].used)} total={disk["/"].total}>
+            {memoryConverter(disk["/"].used)}/{memoryConverter(disk["/"].total)}
+            ({disk["/"].percent.toFixed(1)}%)
+          </PercentageBar>
+        )}
+      </TableCell>
+      <TableCell align="center">{memoryConverter(networkSpeed[0])}/s</TableCell>
+      <TableCell align="center">{memoryConverter(networkSpeed[1])}/s</TableCell>
+      <TableCell>
+        <Link to={`/log/${encodeURIComponent(logUrl)}`}>Log</Link>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+type WorkerRowProps = {
+  /**
+   * Details of the worker
+   */
+  worker: Worker;
+  /**
+   * Detail of the node the worker is inside.
+   */
+  node: NodeDetailExtend;
+};
+
+/**
+ * A single row that represents the data of a Worker
+ */
+const WorkerRow = ({ node, worker }: WorkerRowProps) => {
```

Comment:
We already have a worker table in `dashboard/client/src/components/WorkerTable.tsx`, can we reuse it? Make we should make the UI unified?

Reply:
That table is quite different and it will be difficult to align. One main reason, the table that shows the nodes and worker info together have shared columns, so if a worker row and a node row differs by too much, it will be really confusing to re-use the same column for different information.